src/lib/router.js

```diff
@@ -52,10 +52,23 @@ async function swapContent(url) {
 
 router
   .on("/", async () => {
-    return swapContent("index.html");
+    return swapContent("/");
   })
   .on("/*", async (params) => {
-    return swapContent(params.wild);
+    if (params.wild.endsWith("/index.html")) {
+      // If an internal link refers to "/foo/index.html", strip "index.html" and load.
+      const stripped = params.wild.slice(0, -"index.html".length);
+      return swapContent(`/${stripped}`);
+    } else if (window.location.pathname.endsWith("/")) {
+      // Navaid strips a trailing "/" on its own, so ensure it is added again before loading.
+      return swapContent(`/${params.wild}/`);
+    }
+
+    // Otherwise, this is a request for e.g. "/measure". By calling history.replaceState() to
+    // re-add the trailing slash, Navaid will re-run this code and trigger the latter conditional
+    // above. (If we also call swapContent(), two requests will fire.)
+    // This doesn't modify the back stack, so further navigation works fine.
+    window.history.replaceState(null, null, window.location.pathname + "/");
```

Comment:
For the future...

This makes me think that we're working around navaid enough that it might be worth us either using a different router (page.js maybe?) or just roll our own. Not something we need to do now though.

Reply:
Yes, I agree with this sentiment.